runtime/freelist.c

```diff
@@ -919,12 +915,13 @@ static struct large_free_block *bf_large_least;
 #ifdef HAS_FFS
 /* Nothing to do */
 #elif defined(HAS_BITSCANFORWARD)
+#include <intrin.h>
 static inline int ffs (int x)
 {
   unsigned long index;
   unsigned char result;
   result = _BitScanForward (&index, (unsigned long) x);
-  return (int) index + result;
+  return result ? (int) index + 1 : 0;
```

Comment:
Is the +1 needed? My impression from the docs was that `_BitScanForward` returns a 1-based number like ffs (but I found the docs quite hard to read)

Reply:
I had this in another comment but GitHub seems to have eaten it:

MS's specification doesn't say, but their example says _BitScanForward(12) returns index=2. https://docs.microsoft.com/en-us/cpp/intrinsics/bitscanforward-bitscanforward64?view=vs-2019

I'll run it through precheck anyway.

Reply:
Right, but 12 is 1010 so index=2 is 1-based and doesn't need any +1 to match ffs?

Reply:
I haven't followed much of the discussion but I believe 12 is 1100 in binary. Hope that clarifies the discussion a bit.

Reply:
Hah! I'll learn to count next time. Thanks!